sdk/keyvault/azkeys/testdata/perf/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/perf"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azkeys"
+)
+
+type perfTest struct {
+	client  *azkeys.Client
+	keyName string
+}
+
+func (p *perfTest) Setup() {
+	cred, err := azidentity.NewClientSecretCredential(
+		os.Getenv("AZKEYS_TENANT_ID"),
+		os.Getenv("AZKEYS_CLIENT_ID"),
+		os.Getenv("AZKEYS_CLIENT_SECRET"),
+		nil,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	client, err := azkeys.NewClient(os.Getenv("AZURE_KEYVAULT_URL"), cred, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	p.client = client
+	s := rand.NewSource(time.Now().Unix())
+	p.keyName = fmt.Sprintf("%d", s.Int63())
+
+	_, err = p.client.CreateRSAKey(context.Background(), p.keyName, nil)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (p *perfTest) Run() {
+	_, err := p.client.GetKey(context.Background(), p.keyName, nil)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (p *perfTest) TearDown() {
+	poller, err := p.client.BeginDeleteKey(context.Background(), p.keyName, nil)
+	if err != nil {
+		panic(err)
+	}
+	_, err = poller.PollUntilDone(context.Background(), time.Second)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func main() {
+	perf.RunPerfTest(&perfTest{})
```

Comment:
So, this means that you would be creating one binary for each perf test?  

Typically, on other languages we have one binary per service which defines all the perf tests for that service.
The binary would receive the name of the perf test to execute or list the available tests to run if no test name is enter.

Reply:
I haven't figured this portion out yet. Right now yes it would create a single binary, but I'm working on the single binary that  you describe.